### PR TITLE
[WIP] Add init script

### PR DESCRIPTION
**Pull request by calebeby**
Automatically
- [x] clones boilerplate
- [x] removes .git directory
- [x] git init
- [x] Updates module name in package.json and README.md
- [x] installs dependencies with npm or yarn
- [x] Creates GitHub repo with correct description and homepage url
- [ ] Activates Travis CI (I'm having trouble with Travis API)

I would like to get Travis API dealt with before this is merged.

### CODE REVIEW

File: jst-nuke-init [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      name: 'humanName',
+      message: 'Name of JSTransformer (capitalized properly)'
+    },
+    {
+      name: 'url',
+      message: 'URL to find more about the transformer'
+    },
+    {
+      name: 'username',
+      message: 'GitHub Username (for creating repo)'
```
Comment (RobLoach) [inline jst-nuke-init]: I don't believe a username or password is required to clone the repo. Is this for creating the repo on GitHub? I use SSH for auth, there is no password.

File: jst-nuke-init [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    return answers
+  })
+}
+
+var setup = function (answers) {
+  clone('jstransformers/boilerplate', {dest: answers.transformer}, function () {
+    cd(answers.transformer)
+    run('pwd')
+    rimraf.sync('.git')
+    github.repos.createForOrg({
```
Comment (RobLoach) [inline jst-nuke-init]: http://mikedeboer.github.io/node-github/#api-repos-createForOrg

Seems like a neat idea. Hadn't thought of doing all of this in the command line. Could creating it in the Org be an option we pass? May not want to create a repo for it all the time.

File: jst-nuke-init (hunk 1 above)
Comment (calebeby) [inline jst-nuke-init]: Yes, this creates the repo on GitHub. Would SSH login work with https://github.com/mikedeboer/node-github#authentication? Is that the `userToken`?

File: jst-nuke-init (hunk 2 above)
Comment (calebeby) [inline jst-nuke-init]: Sure, I will make this an option.
`--create-repo`

File: jst-nuke-init [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      name: 'password',
+      type: 'password',
+      message: 'GitHub Password (for creating repo)'
+    }
+  ]).then(function (answers) {
+    github.authenticate({
+      type: "basic",
+      username: answers.username,
+      password: answers.password
+    }, function () {console.log('authenticated with GitHub')})
```
Comment (TimothyGu) [inline jst-nuke-init]: This is not going to work with 2FA (which I use). Check https://github.com/mikedeboer/node-github#creating-a-token-for-your-application for how to create a token and use it here.

Also, please use promises.

File: jst-nuke-init [hunk 4]
```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env node
+
+var exec = require('child_process').execSync
```
Comment (TimothyGu) [inline jst-nuke-init]: Please use ES6 when possible (e.g. `const {execSync} = require('child_process');`, arrow functions etc.).

File: jst-nuke-init (hunk 1 above)
Comment (TimothyGu) [inline jst-nuke-init]: I don't think SSH login would work with the GitHub API, only when pushing through Git.

File: jst-nuke-init [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+      description: `${answers.humanName} support for JSTransformers`,
+      homepage: `https://npmjs.com/${answers.transformer}`,
+      has_wiki: false
+    }, function () {
+      console.log('Finished making repo on GitHub')
+      console.log(`Go to https://travis-ci.org/jstransformers/${answers.transformer} and click enable`)
+    })
+
+    run('git init')
+    run('yarn')
```
Comment (TimothyGu) [inline jst-nuke-init]: How about npm instead?

Comment (calebeby) [inline jst-nuke-init]: OK, I'll set that as the default, and add an option for yarn.

**Comment by RobLoach** [comment]
Nice work :+1:

**Comment by calebeby** [comment]
I think it's good.

**Pull request merged**